x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_form/relevance_tuning_item_content/boosts/boosts.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+
+import { useActions } from 'kea';
+
+import {
+  EuiPageContent,
+  EuiPageContentHeader,
+  EuiPageContentHeaderSection,
+  EuiPageContentBody,
+  EuiTitle,
+  EuiSuperSelect,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { TEXT } from '../../../../../../shared/constants/field_types';
+import { SchemaTypes } from '../../../../../../shared/types';
+
+import { BoostIcon } from '../../../boost_icon';
+import { FUNCTIONAL_DISPLAY, PROXIMITY_DISPLAY, VALUE_DISPLAY } from '../../../constants';
+import { RelevanceTuningLogic } from '../../../relevance_tuning_logic';
+import { Boost, BoostType } from '../../../types';
+
+import { BoostItem } from './boost_item';
+
+import './boosts.scss';
+
+const BASE_OPTIONS = [
+  {
+    value: 'add-boost',
+    inputDisplay: i18n.translate(
+      'xpack.enterpriseSearch.appSearch.engine.relevanceTuning.boosts.addBoostDropDownOptionLabel',
+      {
+        defaultMessage: 'Add boost',
+      }
+    ),
+    disabled: true,
+  },
+  {
+    value: BoostType.Functional,
+    inputDisplay: (
+      <>
+        <BoostIcon type={BoostType.Functional} />
+        {FUNCTIONAL_DISPLAY}
+      </>
+    ),
+  },
+  {
+    value: BoostType.Proximity,
+    inputDisplay: (
+      <>
+        <BoostIcon type={BoostType.Proximity} />
+        {PROXIMITY_DISPLAY}
+      </>
+    ),
+  },
+  {
+    value: BoostType.Value,
+    inputDisplay: (
+      <>
+        <BoostIcon type={BoostType.Value} />
+        {VALUE_DISPLAY}
+      </>
+    ),
+  },
+];
+
+const filterInvalidOptions = (value: BoostType, type: SchemaTypes) => {
+  // Proximity and Functional boost types are not valid for text fields
+  if (type === TEXT && [BoostType.Proximity, BoostType.Functional].includes(value)) return false;
+  return true;
+};
+
+interface Props {
+  name: string;
+  type: SchemaTypes;
+  boosts?: Boost[];
+}
+
+export const Boosts: React.FC<Props> = ({ name, type, boosts = [] }) => {
+  const { addBoost } = useActions(RelevanceTuningLogic);
+
+  const addBoostClick = (value: BoostType) => {
+    addBoost(name, value);
+  };
+
+  const selectOptions = useMemo(
+    () => BASE_OPTIONS.filter((option) => filterInvalidOptions(option.value as BoostType, type)),
+    [type]
+  );
+
+  return (
+    <EuiPageContent className="boosts">
+      <EuiPageContentHeader className="boosts__header" responsive={false}>
+        <EuiPageContentHeaderSection>
```

Comment:
Quick follow up to this from https://github.com/elastic/kibana/pull/90621#discussion_r577057249 - we should likely convert this to generic `EuiPanel` and `EuiFlexGroup` components. I _think_ this will also save us from some EUI CSS overrides (unsetting margin-bottom) as a result.

Reply:
Oh man you're way ahead of me :)